File: java/isbn-verifier/src/main/java/IsbnVerifier.java
public class IsbnVerifier {

    public static String undasher(String code) {
    	return code.replaceAll("\\-", "");
    }	

    public static boolean isValid(String stringToVerify) {
    	int sum = 0;
    	int count = 10;
    	String str = undasher(stringToVerify);

    	if(str.length() != 10 || !str.matches("[0-9]{9}(X|[0-9])$")) {
    		return false;
    	} else {
    		for(String ch: str.split("")) {
    			if(ch.equals("X")) {
    				ch = "10";
    			}
    			sum += (Integer.parseInt(ch) * count);
    			count--;
    		}
    		return (sum % 11) == 0;
    	}
    }
    
}

File: python/isbn-verifier/isbn_verifier.py
import re

def verify(isbn):
    string = undasher(isbn)
    tot = 0
    count = 10
    if not len(string) == 10 or not re.match(r"^[0-9]{9}[0-9|X]{1}$", string):
    	return False
    else:
    	for ch in string:
    		if ch == "X":
    			ch = 10
    		tot += int(ch) * count
    		count -= 1
    	return (tot % 11) == 0

def undasher(code):
	return code.replace("-", "")
